### PR TITLE
Start working on Wizard

### PR DESCRIPTION
**Pull request by Keno**
Not nearly complete, but something to start from.

Demo: https://asciinema.org/a/7gDelUUX0wYtmgO6V2Behkmec

### CODE REVIEW

File: src/Wizard.jl [hunk 1]
```diff
@@ -0,0 +1,207 @@
...
+
+    println()
+
+    build_path = tempname()
+    mkpath(build_path)
+    local history
+    cd(build_path) do
+        temp_prefix() do prefix
+            histfile = joinpath(build_path, ".bash_history")
+            dr = DockerRunner(prefix = prefix, platform = :linux64,
```
Comment (ararslan) [inline src/Wizard.jl]: Platform arguments should now be using `Platform` objects. In this case it should be `platform = Linux(:x86_64)`.

Comment (Keno) [reply src/Wizard.jl]: True, there's a bunch of other code like this in this package also.

Comment (ararslan) [reply src/Wizard.jl]: Ah yeah you're right. This is Binary**Builder**, not Binary**Provider**. Perhaps I'll go through and update things here.

Comment (Keno) [reply src/Wizard.jl]: I already have those changes locally, so if you haven't yet, don't start. Otherwise feel free to put up a PR.

Comment (ararslan) [reply src/Wizard.jl]: Okay, I just did. Didn't see your message until just now.

File: src/Wizard.jl [hunk 2]
```diff
@@ -0,0 +1,371 @@
...
+            print_with_color(:bold, "\n\t\t\tBuild complete. Analyzing...\n\n")
+
+            audit(prefix; platform=Windows(:x86_64), verbose=true, autofix=false)
+
+            match_files(prefix, Windows(:x86_64), state.files)
+        end
+    end
+
+    println("")
+    println("You have successfully built for Win64. Congratualations!")
```
Comment (ararslan) [inline src/Wizard.jl]: > Congratulations

You've got an excess a in there

**Comment by vchuravy** [comment]
While playing around with this branch I encountered the following issues.

- `tar.bz2` support missing
-  `deps/Makefile` isn't run automatically
-  `deps/sandbox_root` is missing on a fresh checkout
- Can't write to `/tmp`
-  `bash: /root/.bashrc: Permission denied` when starting the wizard

**Comment by Keno** [comment]
FWIW, https://github.com/JuliaPackaging/BinaryBuilder.jl/pull/34 should address the last of @vchuravy issues (the sandbox_root one).

**Pull request merged**